scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Split.scala

```diff
@@ -142,6 +142,12 @@ case class Split(
         case x => copy(indents = Indent(x, expire, when) +: indents)
       }
 
+  def withIndent(indent: Indent[Length]): Split =
```

Comment:
i'd switch the implementation with the one just above. why deconstruct `Indent` object, only to put back again?

Reply:
Because I want to keep laziness in argument computation of the above method. Yes, it's a little bit inefficient but I don't know how to
1. keep lazy arguments version
2. add method which receives constructed `Indent`

without code duplication.

Reply:
I see your solution. We can merge https://github.com/scalameta/scalafmt/pull/1796 first and then I'll reuse it here

Reply:
i had the same exact need a couple of days ago, solved it like this:

https://github.com/scalameta/scalafmt/pull/1796/commits/c05e44f5e54a2d27c6ce2a0d465e652633758641#diff-64605fc3e29e25f92371a46c89a960c7R161